painters/model_def_standard.py

```diff
@@ -0,0 +1,134 @@
+import os
+
+import keras
+import numpy as np
+import tensorflow as tf
+from keras import backend as K
+from keras.datasets.cifar import load_batch
+from keras.layers import (Activation, Conv2D, Dense, Dropout, Flatten,
+                          MaxPooling2D)
+from keras.losses import categorical_crossentropy
+from keras.metrics import categorical_accuracy
+from keras.models import Sequential
+from keras.optimizers import Adam
+from keras.utils.data_utils import get_file
+
+import pedl
+from pedl.data import ArrayBatchLoader
+from pedl.frameworks.keras import KerasTrial
+from pedl.frameworks.keras.data import KerasDataAdapter
+from pedl.frameworks.util import elementwise_mean
+from sklearn.metrics import roc_auc_score
+
+from train_cnn import _cnn, IMGS_DIM_3D, BATCH_SIZE, SOFTMAX_SIZE
+from data_provider import load_organized_data_info, train_val_dirs_generators
+from validation import _create_pairs_generator, IMGS_DIM_1D
+from utils import pairs_dot
+
+import objgraph
+
+def categorical_error(y_true, y_pred):
+    return 1. - categorical_accuracy(y_true, y_pred)
+
+
+class PainterTrial(KerasTrial):
+    def __init__(self, hparams):
+        super().__init__(hparams)
+
+        self.kernel_size = pedl.get_hyperparameter("kernel_size")
+        self.dropout = pedl.get_hyperparameter("dropout")
+        self.pool_size = pedl.get_hyperparameter("pool_size")
+        self.L2_REG = pedl.get_hyperparameter("L2_REG")
+        self.lr = pedl.get_hyperparameter("lr")
+        self.my_batch_size = pedl.get_hyperparameter("batch_size")
+        self.data_info = load_organized_data_info(IMGS_DIM_1D)
+
+    # def session_config(self, hparams):
+    #     if hparams.get("disable_CPU_parallelism", False):
+    #         return tf.ConfigProto(intra_op_parallelism_threads=1,
+    #                               inter_op_parallelism_threads=1)
+    #     else:
+    #         return tf.ConfigProto()
+
+    def build_model(self, hparams):
+        model = _cnn(IMGS_DIM_3D)
+        return model
+
+    def optimizer(self):
+        adam = Adam(lr=self.lr)
+        return adam
+
+    def loss(self):
+        return categorical_crossentropy
+
+    def batch_size(self):
+        return self.my_batch_size
+
+    def roc_auc(self, y_true, y_pred):
+        pass
+
+    def validation_metrics(self):
+        return {}
```

Comment:
Leave a comment on why you are defining this function. I assume it's because this is an abstract function and the trial class fails instantiation if it isn't defined, but it isn't immediately clear from this code.